xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEStream.cpp

```diff
@@ -444,6 +447,26 @@ bool CActiveAEStream::SetResampleRatio(double ratio)
   return true;
 }
 
+void CActiveAEStream::GetFFmpegInfo(int &profile, enum AVMatrixEncoding &matrix_encoding, enum AVAudioServiceType &audio_service_type)
+{
+  profile = m_profile;
+  matrix_encoding = m_matrixEncoding;
+  audio_service_type = m_audioServiceType;
+}
+
+void CActiveAEStream::SetFFmpegInfo(int profile, enum AVMatrixEncoding matrix_encoding, enum AVAudioServiceType audio_service_type)
+{
+  if (m_profile != profile ||
+      m_matrixEncoding != matrix_encoding ||
+      m_audioServiceType != audio_service_type)
+  {
+    m_profile = profile;
+    m_matrixEncoding = matrix_encoding;
+    m_audioServiceType = audio_service_type;
```

Comment:
Currently only on the init of dsp used. But not sure how this data is handled inside stream and maybe change on the fly? Thats why I added this way to make it more easy behind, then only a extra code must be included on addon and engine but nowhere a change of the function values is required.